packages/sdk/src/client.ts

```diff
@@ -237,13 +237,30 @@ export function SoundClient({
       }
     }
 
-    const remaining =
+    // Get the edition's remaining token quantity.
+    const { signerOrProvider } = await _requireSignerOrProvider()
+    const editionContract = SoundEditionV1_1__factory.connect(mintSchedule.editionAddress, signerOrProvider)
+    const editionTotalMinted = (await editionContract.totalMinted()).toNumber()
+    const editionMaxMintable = await editionContract.editionMaxMintable()
+    const editionRemainingQty = editionMaxMintable - editionTotalMinted
+
+    if (!editionRemainingQty) {
+      return 0
+    }
+
+    // Get eligible quantity for the user on this mint schedule.
+    const remainingForSchedule =
       (typeof mintSchedule.maxMintable === 'function'
         ? mintSchedule.maxMintable(timestamp)
         : mintSchedule.maxMintable) - mintSchedule.totalMinted
 
-    const alreadyMinted = await numberMinted({ editionAddress: mintSchedule.editionAddress, userAddress })
-    return Math.max(Math.min(remaining, mintSchedule.maxMintablePerAccount - alreadyMinted), 0)
+    const mintedByUserFromSchedule = await numberMinted({ editionAddress: mintSchedule.editionAddress, userAddress })
+    const eligibleForUserOnSchedule = mintSchedule.maxMintablePerAccount - mintedByUserFromSchedule
+    const scheduleEligibleQty = Math.min(remainingForSchedule, eligibleForUserOnSchedule)
+
+    // Return the minimum of the two. The number of tokens minted within the mint schedule
```

Comment:
Promise.all on all the stuff that doesn't need previous data to be called

Reply:
I was only able to combine 2 calls. I figure `numberMinted()` shouldn't be called until after we check if the edition is sold out since we potentially return early. https://github.com/soundxyz/sdk/pull/168/commits/d934a5f9dff220ad74011fd82949ddc5e598990b